File: questions/728/python/Solution.py
class Solution(object):
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        Ans = []
        for num in range(left,right+1):
            eligible = True
            currentNum = num;
            while(num > 0):
                digit = num % 10
                if digit == 0 or currentNum % digit > 0:
                    eligible = False
                num = num / 10
            if eligible == True:
                Ans.append(currentNum)
        return Ans

File: questions/728/java/Solution.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> answerList = new ArrayList<>();
        for(int i=left;i<=right;i++){
            int num = i;
            boolean isEligible = true;
            while(i>0){
                int digit =0;
                digit = i %10;
                if (digit ==0 || num % digit >0){
                    isEligible = false;
                }
                i = i/10;
            }
            i=num;
            if (isEligible){
                answerList.add(num);
            }  
        }
        return answerList; 
    }
}